### DISCUSSION TITLE
Error: "proto: unknown field 'resources'" When Sending Request via Script

### DISCUSSION THREAD

**Discussion by Bouba-Barry** [cerbos/cerbos#2396] [Q&A] [answered]
Hi Cerbos Community,

I'm encountering an issue while testing a Cerbos PDP request. Here's the scenario:

When I send the same JSON input via a browser (using a REST client or similar) to my local Cerbos server running on http://localhost:3592, the request works perfectly, and I get the expected response.
However, when I send the same JSON input via my script (executed in VS Code), I get the following error:

{"code":3,"message":"proto: (line 10:5): unknown field \"resources\""}

Here is the content of my JSON input file:

```json
{
   "requestId":"test",
   "principal":{
      "id":"drmar",
      "scope":"acme.corp",
      "roles":[
         "medecin"
      ],
      "attr":{
         "affectation":[
            "Aile_Est",
            "Aile_Ouest"
         ],
         "specialites":[
            "CARDIOLOGIE"
         ]
      }
   },
   "resources":[
      {
         "resource":{
            "id":"dm0001",
            "kind":"dm",
            "policyVersion":"20210210",
            "scope":"acme.corp",
            "attr":{
               "desc":"Consultation cardiologie",
               "status":"clos",
               "medecin":"drlam",
               "service":"Cardiologie",
               "specialite":"CARDIOLOGIE"
            }
         },
         "actions":[
            "lire_dm"
         ]
      },
      {
         "resource":{
            "id":"dm0002",
            "kind":"dm",
            "scope":"acme.corp",
            "attr":{
               "desc":"Opération appendicite",
               "status":"clos",
               "medecin":"drmar",
               "service":"Chirurgie",
               "specialite":"CHIRURGIE"
            }
         },
         "actions":[
            "lire_dm",
            "archiver_dm"
         ]
      }
   ],
   "includeMeta":true
}
```

Notes:

The Cerbos server is running locally on http://localhost:3592.
The JSON is valid (I double-checked the syntax).
The request works when sent directly via a browser or Postman with the same JSON.
Could this be related to the way the script processes or sends the JSON file? Is there any special requirement for the resources field in the payload?

Thanks in advance for your help!

**Comment by oguzhand95** [comment]
Hi @Bouba-Barry,

Could you share the snippet of code building and sending this request?

**Comment by Bouba-Barry** [comment, marked as answer]
Hi,

I realized that the issue was with the URL I used to send the request, not my input.json. I mistakenly omitted the /resources endpoint in the URL. I didn’t catch this earlier because it didn’t result in a 404 error, and I’m not very familiar with the specific error code 3 I received.

Now that I’ve fixed it, it’s working perfectly. Thank you for your help!

For reference, here’s the script I’m using:
input_file="$(pwd)/tests/input.json"


if [[ ! -f "$input_file" ]]; then
  echo "Erreur : Le fichier $input_file n'existe pas."
  exit 1
fi

curl -X POST http://localhost:3592/api/check/resources \
  -H "Content-Type: application/json" \
  --data @"$input_file"
